crates/spk-solve/src/io.rs

```diff
@@ -193,17 +195,167 @@ where
+        self.show_unresolved_requests(state);
+        self.show_var_requests(state);
+        self.show_options(state);
+    }
+
+    fn show_full_menu(&self, prompt_prefix: &str) {
+        println!("{} Enter a letter for an action:", prompt_prefix.yellow());
+        println!(" ? - Print help (these details)");
+        println!(" r - Show resolved packages");
+        println!(" u - Show unresolved requests");
+        println!(" v - Show var requests");
+        println!(" o - Show options");
+        println!(" s, a - Show state [all of the above]");
+        println!(" c - Run solver to completion, removes step/stop");
+        println!(" Ctrl-c - Interrupt this program");
+        println!(" any other - Continue solving");
+    }
+
+    fn remove_step_and_stop_setting(&mut self) {
+        self.settings.stop_on_block = false;
+        self.settings.step_on_block = false;
+        self.settings.step_on_decision = false;
+    }
+
+    fn show_state_menu(
+        &mut self,
+        current_state: &Option<Arc<State>>,
+        prompt_prefix: String,
+    ) -> Result<()> {
+        // TODO: change the timeout that auto-increases the verbosity
+        // or disable when this menu is active
+        if let Some(state) = current_state {
+            // Simplistic menu for now
+            let mut done = false;
+            while !done {
+                // Show a compressed version of the menu
+                print!(
+                    "{} Select one of [r,u,v,o,s,a,c,?,C-c]> ",
+                    prompt_prefix.yellow()
+                );
+
+                // Get selection
+                let response = self.wait_for_user_selection()?;
+                let selection = match response.to_lowercase().chars().next() {
+                    None => continue,
+                    Some(c) => c,
+                };
+
+                // Act on the selection
+                match selection {
+                    '?' => self.show_full_menu(&prompt_prefix),
+                    'r' => self.show_resolved_packages(state),
+                    'u' => self.show_unresolved_requests(state),
+                    'v' => self.show_var_requests(state),
+                    'o' => self.show_options(state),
+                    's' | 'a' => self.show_state(state),
+                    'c' => {
+                        self.remove_step_and_stop_setting();
+                        done = true;
+                    }
+                    // TODO: could look at adding other things in future:
+                    // - show dep graph image based on current resolved/unresolved
+                    // - a breakpoint for a request, with  continue till such request
+                    // - launch spk env based on current resolved packages
+                    // - rewind the solve
+                    // - save/restore point for the solve, for use in tests
+                    _ => done = true,
+                }
+            }
```

Comment:
```suggestion
            loop {
                // Show a compressed version of the menu
                print!(
                    "{} Select one of [r,u,v,o,s,a,c,?,C-c]> ",
                    prompt_prefix.yellow()
                );

                // Get selection
                let response = self.wait_for_user_selection()?;
                let selection = match response.to_lowercase().chars().next() {
                    None => continue,
                    Some(c) => c,
                };

                // Act on the selection
                match selection {
                    '?' => self.show_full_menu(&prompt_prefix),
                    'r' => self.show_resolved_packages(state),
                    'u' => self.show_unresolved_requests(state),
                    'v' => self.show_var_requests(state),
                    'o' => self.show_options(state),
                    's' | 'a' => self.show_state(state),
                    'c' => {
                        self.remove_step_and_stop_setting();
                        break;
                    }
                    // TODO: could look at adding other things in future:
                    // - show dep graph image based on current resolved/unresolved
                    // - a breakpoint for a request, with  continue till such request
                    // - launch spk env based on current resolved packages
                    // - rewind the solve
                    // - save/restore point for the solve, for use in tests
                    _ => break,
                }
            }
```

I don't see a need for `done` here.

Reply:
Rebased and swapped the while for a loop.